tests/unit/multiclass/tree/RandomForest_unittest.cc

```diff
@@ -279,3 +283,69 @@ TEST(RandomForest,classify_non_nominal_test)
 	SG_UNREF(c);
 	SG_UNREF(eval);
 }
+
+TEST(RandomForest, test_scores)
+{
+	// Test data generated using script
+	// https://gist.github.com/olinguyen/d93cd4222b404d26dc859f100d700be6
+	// Generates data for y = (x1 v x2) < 5 as decision boundary
+	sg_rand->set_seed(1);
+	float64_t data_A[] = {9, 7, 5, 5, 6, 7, 8, 9, 6, 8,
+	                      1, 4, 2, 1, 3, 2, 1, 3, 1, 0};
+
+	SGMatrix<float64_t> data(data_A, 2, 10, false);
+
+	CDenseFeatures<float64_t>* features_train =
+	    new CDenseFeatures<float64_t>(data);
+
+	float64_t labels[] = {0.0, 0.0, 0.0, 0.0, 0.0, 1.0, 1.0, 1.0, 1.0, 1.0};
+	SGVector<float64_t> lab(labels, 10);
+	CMulticlassLabels* labels_train = new CMulticlassLabels(lab);
+
+	float64_t test_A[] = {9, 7, 5, 5, 6, 7, 8, 9, 6, 8,
+	                      1, 0, 0, 1, 1, 1, 1, 2, 0, 0};
+
+	SGMatrix<float64_t> test_data(test_A, 2, 10, false);
+
+	CDenseFeatures<float64_t>* features_test =
+	    new CDenseFeatures<float64_t>(test_data);
+
+	CRandomForest* c = new CRandomForest(features_train, labels_train, 10, 2);
+	SGVector<bool> ft = SGVector<bool>(2);
+	ft[0] = false;
+	ft[1] = false;
+	c->set_feature_types(ft);
+
+	CMeanRule* mr = new CMeanRule();
+	c->set_combination_rule(mr);
+	c->parallel->set_num_threads(1);
+	c->train(features_train);
+
+	auto result = c->apply_binary(features_test);
+	SGVector<float64_t> res_vector = result->get_labels();
+	SGVector<float64_t> values_vector = result->get_values();
+
+	EXPECT_EQ(-1.0, res_vector[0]);
```

Comment:
so I guess these come from your manually created function and this just asserts that the RF is working correctly, right?